lib/puppet/pops/time/timespan.rb

```diff
@@ -671,7 +711,7 @@ def internal_parse(str)
     end
 
     class Format
-      DEFAULTS = ['%D-%H:%M:%S', '%D-%H:%M', '%H:%M:%S', '%H:%M'].map { |str| FormatParser.singleton.parse_format(str) }
+      DEFAULTS = ['%D-%H:%M:%S.%-N', '%H:%M:%S.%-N', '%M:%S.%-N', '%S.%-N', '%D-%H:%M:%S', '%H:%M:%S', '%D-%H:%M', '%S'].map { |str| FormatParser.singleton.parse_format(str) }
```

Comment:
This needs to be in the spec as well. IIRC it now has the earlier version of default formats.